File: Reverse Integer/ReverseV1.java
class Solution {
    public int ReverseV1(int x) {
        StringBuilder num = new StringBuilder(""+x);
        num.reverse();
        int ans = 0;
        if(num.indexOf("-") != -1){
            num.deleteCharAt(num.indexOf("-"));
            num.insert(0,"-");
        }
        
        long res = Long.parseLong(num.toString());
        ans = (int) res;
        if(res > Integer.MAX_VALUE || res < Integer.MIN_VALUE)
            ans = 0;
        return ans;
        
    }
}

File: Reverse Integer/reverseV1.py
class Solution:
    def reverse(self, x: int) -> int:
        s1 = str(x)[::-1]
        if('-' in s1):
            s1 = s1.replace('-','')
            s1 = '-'+s1
        num = int(s1)
        return num if -2**31 < num < 2**31-1 else 0    